### PR TITLE
bpo-32972: Add coroutine support to unittest

### PR DESCRIPTION
**Pull request by johnandersen777**
https://bugs.python.org/issue32972

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Review by 1st1 — CHANGES_REQUESTED** [review]
This needs more discussion in https://bugs.python.org/issue32972

File: Lib/unittest/case.py [hunk 1]
```diff
@@ -1393,6 +1400,34 @@ def shortDescription(self):
...
+    if sys.version_info >= (3, 7):
+        coroutineRunner = asyncio.run
+    else:
+        coroutineRunner = coro_runner
+
+    @classmethod
+    def runMethod(cls, meth):
+        """Executes a function or coroutine function using the
+        coroutineRunner."""
+        if inspect.iscoroutinefunction(meth):
```
Comment (PetterS) [inline Lib/unittest/case.py:1425]: This does not work if the method is decorated with @unittest.mock.patch.

**Review by PetterS — COMMENTED** [review]
n/a

**Comment by ned-deily** [comment]
@pdxjohnny Sorry that the discussion about this issue has languished, especially since the submitting of an alternate PR.  However, I see your PR was created against the 3.6 branch; for enhancements, we require that PRs be against the "master" branch.  So I'm going to close this PR and you can decide how you want to proceed, either submitting an updated PR against "master" or perhaps commenting on the other open PR.  In any case, thanks for getting the discussion started and submitting this PR!

https://devguide.python.org/pullrequest/

**Comment by pdxjohnny** [comment]
I'm glad there has been discussion about it. Thank you!

**Pull request closed without merging**